content/dev/blog/analyzing-so-data.Rmd

```diff
@@ -0,0 +1,308 @@
+easier to work with these kinds of data structures, but going from a raw CSV to a well-labeled, cleaned dataset can take some work.
+
+However, with some decent data wrangling skills, this process can be scripted efficiently. In this post, our R developers discuss how they used Crunch's R packages to harness the flexibility of Crunch's APIs along with the power of R to script repetitive cleaning tasks, reshape the data in ways that the web application doesn't let you do, and use statistical modeling to correct biases in the underlying data. We go from a raw CSV of values to a polished dataset that nontechnical users can explore and gain insights from on the web, while still maintaining the ability to do any statistical analysis we want in R.
+
+```{r include=FALSE}
+crunch::login()
+```
+
+# 1. Upload the data
+
+To begin, let's load the [crunch](http://crunch.io/r/crunch/) package along with some others we'll use in the data processing.
+
+```{r library load, message = FALSE}
+library(crunch)
+library(stringr)
+library(tidyverse)
+```
+
+Now, read in the CSVs from the survey and upload the data to the Crunch platform.
+
+```{r Read data, results=FALSE, message=FALSE, cache = TRUE}
+stack_df <- read_csv("survey_results_public.csv", progress = FALSE)
+schema <- read_csv("survey_results_schema.csv", progress = FALSE)
+```
+
+```{r Upload to crunch, results = FALSE, message = FALSE, cache = TRUE}
+ds <- newDataset(stack_df, name = "StackOverflow Developer Survey 2017")
+```
+
+In the Crunch web app, we can take a look at some variable summaries to start to get a sense of the data structure.
+
+```{r echo=FALSE, out.width='100%'}
+knitr::include_graphics('./images/so_var_cards.png')
+```
+
+The StackOverflow questions are stored in two distinct formats. There's a human
+readable version, which is presented to the respondent, and a machine readable
+one which is stored as the variable names. One of the first, and most frequent
+questions that you'll likely ask when analyzing this survey is "what was that
+question again?". The StackOverflow team has helpfully provided the survey
+schema and questionnaire which shows the relationship between the variable names
+and the questions, but there's not a natural way to store this information in
+either a csv file or an R dataframe. Storing the questions in the first row of
+the table is not a great idea, so you will probably end up just keeping the
+schema and questionnaire in separate files and cross-referencing them when
+needed.
+
+Crunch allows you to store more information than just the name about the
```

Comment:
Is this all of the metadata we brought in? Any other steps that Mike did to polish using the questionnaire? Discuss this a bit more.